lib/src/workspace.rs

```diff
@@ -299,8 +299,8 @@ impl Workspace {
         &self.working_copy
     }
 
-    pub fn working_copy_mut(&mut self) -> &mut LocalWorkingCopy {
-        &mut self.working_copy
+    pub fn set_working_copy(&mut self, working_copy: LocalWorkingCopy) {
+        self.working_copy = working_copy
```

Comment:
Another idea is to add a wrapper that holds `&mut Workspace` and update `wrapper.ws.working_copy` on finish.

```rust
let wrapper = ws.start_working_copy_mutation();
wrapper.locked_wc()... // &mut dyn LockedWorkingCopy
wrapper.finish() // wrapper.ws.working_copy = locked_wc.finish()
```

Reply:
Ah, much better. Done.